### ISSUE TITLE
Issues from Mark for final rho release

### ISSUE THREAD

**Issue by noahl** [quipucords/rho#537]
This is from a conversation we had on Tuesday 12/5.

- [x] Summary fact for Fuse (#538 ).
- [x] Summary fact for BRMS (#539 ).
- [x] Issue #477 .
- [x] Issue #284 .
- [x] Let Mark take a look at the release and sign off (#540 )

**Comment by noahl** [comment]
#477 is mitigated by Elijah's work.

**Comment by noahl** [comment]
Added two new issues so that every element of the list now has its own issue.

**Comment by noahl** [comment]
All issues are now complete except for getting Mark to sign off.

**Comment by kdelee** [comment]
Did the ClientAlive options end up merged into master that Chris talked
about in this comment
https://github.com/quipucords/rho/issues/477#issuecomment-349731629 ?

**Comment by noahl** [comment]
@kdelee I don't think so. I saw those, but it looked like we need to do some more research and testing before we can turn those on, and I am reluctant to wait on that for the rho release.

**Comment by noahl** [comment]
Overall, I know #477 is still open, but I think we should go ahead and release for two reasons.
1. #477 was open for previous rho releases too, so we're not regressing anything. In fact, we're making incremental progress towards fixing it, and getting that to our users should benefit them.
2. #477 is broad. I don't see how we can guarantee that we will *never* hang. All we can do is keep incrementally fixing individual causes of hangs as we find them, and that's what we've done in this release.

With that said, if you want to do the testing and send our a PR for the `ClientAlive` options, by all means, go ahead.

@chambridge what do you think of this? Should we wait?

**Comment by kdelee** [comment]
On Dec 13, 2017 1:17 PM, "noahl" <notifications@github.com> wrote:

Overall, I know #477 <https://github.com/quipucords/rho/issues/477> is
still open, but I think we should go ahead and release for two reasons.

   1. #477 <https://github.com/quipucords/rho/issues/477> was open for
   previous rho releases too, so we're not regressing anything. In fact, we're
   making incremental progress towards fixing it, and getting that to our
   users should benefit them.
   2. #477 <https://github.com/quipucords/rho/issues/477> is broad. I don't
   see how we can guarantee that we will *never* hang. All we can do is
   keep incrementally fixing individual causes of hangs as we find them, and
   that's what we've done in this release.

With that said, if you want to do the testing and send our a PR for the
ClientAlive options, by all means, go ahead.

@chambridge <https://github.com/chambridge> what do you think of this?
Should we wait?


I'm chiming in from PTO, i.e. don't have the time allocated to test/make a
PR myself.

Just wasn't sure where that landed and what the results of Chris's
experimentation were.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/quipucords/rho/issues/537#issuecomment-351476517>, or mute
the thread

**Comment by mdvickst** [comment]
I'd like to consider the suggestion on ClientAlive for the 31 release. If .31 is truly going to be the last release of rho before sonar it would be good to have ClientAlive plus https://github.com/quipucords/rho/issues/504 (a longer timeout for active commands that are running) fixed. I feel like these would go a long way towards the stability of large scans since Rho will inevitably hit a host which either doesn't close out it's ssh connection or a command hangs etc. It's really painful for a large scan to hang because 1 task on 1 host failed.
